### ISSUE TITLE
New data structure for messages

### ISSUE THREAD

**Issue by koppor** [InAnYan/jabref#90]
(Just as note - to be refined / implemented)

Currently, the `BibEntryChatHistory` seems to have the chat history for one entry, tight to one bibdatabase, which is held in a map.

There is a single map for one JabRef instance - independent of the number of libraries opened.

Since the chats of entries of one library are NOT realted to chats to entries of other libraries, one can simplify the datastructure as sketeched below:

```java
    /**
     * Data structure:
     *
     * A) path -> {BibEntry}          // BibEntries are not orderded, therefore a set
     * B)   BibEntry -> [ChatMessage] // CHat Messages are in order - thus, a list
     *
     * A) is done by different MVStore maps - having the path as name
     * B) is done by MVMaps
     */
    private final Map<String, List<MessageData>> bibDatabaseChatHistoryMap;
```

I need leave now, I plan to resume work on this on Thursday the latest.

**Comment by koppor** [comment]
Does not *fully" work. See https://github.com/koppor/mvstore-mwe/pull/1 for details.

I still think, we should have one map per BibDatabaseContext.

**Comment by InAnYan** [comment]
Oliver, I remember you reworked the chat history.

Does this issue still relevant now?

Oh, and also, don't link implementation with `BibEntry` or `BibDatabase` or `BibDatabaseContext`, they are unstable runtime objects. The only "stability" is file paths and citation keys (if they are right). That's why I always so dependent on them

**Comment by koppor** [comment]
> Oliver, I remember you reworked the chat history.
> 
> Does this issue still relevant now?

No. I will close. The issue is fixed at Yes, fixed with [JabRef/jabref@`0a33b34` (#11430)](https://github.com/JabRef/jabref/pull/11430/commits/0a33b343a3532855013411ca6d02e9d5dea8c640).

> Oh, and also, don't link implementation with `BibEntry` or `BibDatabase` or `BibDatabaseContext`, they are unstable runtime objects. The only "stability" is file paths and citation keys (if they are right). That's why I always so dependent on them

I know and I did.
